frontend/awx/administration/execution-environments/ExecutionEnvironmentPage/ExecutionEnvironmentDetails.tsx

```diff
@@ -0,0 +1,95 @@
+import { Label } from '@patternfly/react-core';
+import { useTranslation } from 'react-i18next';
+import { useParams } from 'react-router-dom';
+import {
+  DateTimeCell,
+  PageDetail,
+  PageDetails,
+  TextCell,
+  useGetPageUrl,
+  usePageNavigate,
+} from '../../../../../framework';
+import { useGetItem } from '../../../../common/crud/useGet';
+import { LastModifiedPageDetail } from '../../../../common/LastModifiedPageDetail';
+import { awxAPI } from '../../../common/api/awx-utils';
+import { ExecutionEnvironment } from '../../../interfaces/ExecutionEnvironment';
+import { AwxRoute } from '../../../main/AwxRoutes';
+
+export function ExecutionEnvironmentDetails() {
+  const params = useParams<{ id: string }>();
+  const { data: execution_environment } = useGetItem<ExecutionEnvironment>(
+    awxAPI`/execution_environments/`,
+    params.id
+  );
+  return execution_environment ? (
+    <ExecutionEnvironmentDetailInner execution_env={execution_environment} />
+  ) : null;
+}
+
+export function ExecutionEnvironmentDetailInner(props: { execution_env: ExecutionEnvironment }) {
+  const { t } = useTranslation();
+  const getPageUrl = useGetPageUrl();
+  const pageNavigate = usePageNavigate();
+
+  const execution_env = props.execution_env;
+
+  return (
+    <PageDetails data-cy="execution-environment-page-detail">
+      <PageDetail data-cy="execution-environment-page-detail" label={t('Name')}>
```

Comment:
can this data-cy value be 'execution-environment-name' instead? I see that 'execution-environment-page-detail is repeated several times in this file- Cypress will get confused and not know which one to click on if we use it as a locator. They all need to be unique.

Reply:
done